language/move-prover/bytecode-to-boogie/src/spec_translator.rs

```diff
@@ -860,14 +862,45 @@ impl<'env> SpecTranslator<'env> {
                     res = format!("Dereference(__m, {})", res);
                     t = *vt;
                 }
+
                 for f_or_i in fields_and_indices {
-                    if let FieldOrIndex::Field(f) = f_or_i {
-                        let (s, tf) = self.translate_field_access(&t, f);
-                        res = format!("SelectField({}, {})", res, s);
-                        t = tf;
-                    } else {
-                        // TODO: Implement vector index.  For now, generates illegal Boogie.
-                        res = format!("Unimplemented vector index. AST = {:?}", f_or_i);
+                    match f_or_i {
+                        FieldOrIndex::Field(f) => {
+                            let (s, tf) = self.translate_field_access(&t, f);
+                            res = format!("SelectField({}, {})", res, s);
+                            t = tf;
+                        }
+                        FieldOrIndex::Index(i) => {
+                            match i {
+                                SpecExp::Update(ind, val) => {
+                                    let BoogieExpr(idx, _t_ind) = self.translate_expr( & ind);
```

Comment:
Same here -- Rust style is to use `_`.

Reply:
I left `_t_ind` to mention it in a TODO comment for type checking later.